relayer/src/source.ts

```diff
@@ -78,35 +91,59 @@ class Source {
       .pipe(concatMap(({ paraId, paraHead }) => {
         const parachain = this.parachainsMap.get(paraId);
         if (!parachain) throw new Error(`Uknown paraId: ${paraId}`);
+
+        const { feedId, chain, signer } = parachain;
+
         return parachain.fetchParaBlock(paraHead)
-          .pipe(map((block) => this.addBlockMetadata({
-            block,
-            hash: paraHead,
-            feedId: parachain.feedId,
-            chain: parachain.chain,
-          })));
+          .pipe(map(({ block }) => {
+            const blockStr = JSON.stringify(block);
+            const number = block.header.number.toString();
```

Comment:
Why does number need to be converted into a string though?

Reply:
in this case number is `Compact<BlockNumber>`, so we have to convert it to number or string before submission anyway

Reply:
Hm, I think number would be better. By default Substrate uses `u32` anyway and JS can store up to 53-bit integers without loss, so it should be fine, otherwise `BigInt` can be used. It is just strange to see strings for block number.